analyzer/codechecker_analyzer/arg.py

```diff
@@ -36,6 +36,36 @@ def __call__(self, parser, namespace, value, option_string=None):
         if 'ordered_checkers' not in namespace:
             namespace.ordered_checkers = []
         ordered_checkers = namespace.ordered_checkers
+        # Map each checker to whether its enabled or not.
         ordered_checkers.append((value, self.dest == 'enable'))
 
         namespace.ordered_checkers = ordered_checkers
+
+
+class OrderedAnalyzerConfigAction(argparse.Action):
+    """
+    Action to store --analyzer-config values. These may come from many sources,
+    including the CLI, the CodeChecker config file, the saargs file and the
+    tidyargs file, or some other places.
+    """
+
+    def __init__(self, option_strings, dest, nargs=None, **kwargs):
+        if nargs is not '*':
+            raise ValueError("nargs must be '*' for backward compatibility "
+                             "reasons!")
+        super(OrderedAnalyzerConfigAction, self).__init__(option_strings, dest,
+                                                          nargs, **kwargs)
+
+    def __call__(self, parser, namespace, value, option_string=None):
+
+        if 'analyzer_config' not in namespace:
+            namespace.analyzer_config = []
+        analyzer_config = namespace.analyzer_config
+
+        assert isinstance(value, list), \
+               "--analyzer-config value  (" + str(value) + ") is not a list," \
+               " but should be if nargs is not None!"
```

Comment:
Might be useful to know that strings can be assembled simpler:
```suggestion
               f"--analyzer-config value  ({value}) is not a list," \
               " but should be if nargs is not None!"
```